compiler/rustc_codegen_llvm/src/debuginfo/metadata.rs

```diff
@@ -1070,6 +1071,16 @@ fn build_struct_type_di_node<'ll, 'tcx>(
         None
     };
 
+    if find_attr!(cx.tcx.get_all_attrs(adt_def.did()), AttributeKind::DebuginfoTransparent(..)) {
+        let ty = struct_type_and_layout.non_1zst_field(cx).unwrap().1.ty;
```

Comment:
Is there a way to ensure that `#[debuginfo_transparent]` can only be used when `#[repr(transparent)]` is also used? And how should I handle the case where all fields are 1zst's?

Reply:
I've implemented a restriction to `#[repr(transparent)]` types.